### PR TITLE
Tutorial on the different Series Types

### PR DESCRIPTION
**Pull request by henrifroese**
Only a first draft at the moment; critique welcome.

Focus on / extended explanation of Representation Series

Question: Is this the correct location for the file (`website/docs/`)?

### CODE REVIEW

**Comment by jbesomi** [comment]
> Only a first draft at the moment; critique welcome (you can have a look at it e.g. by pasting everything in here.

You can directly add it to the website so that we can see it by clicking "view deployment".

Also, "view file" do the job: [view file](https://github.com/jbesomi/texthero/blob/5fb3551fb7a4f6eb6d1d13d5e5d2c35a32d7616d/website/docs/different_series_types_tutorial.md).

> Question: Is this the correct location for the file (website/docs/)?

See #94 for a better contextualization. The idea is to have this file under the "Getting Started" tab.

We might want to call this file: `getting-started-heroseries.md`


Review:

1. Very good start 💪 
1. The style should be aligned somehow with the [getting-started](https://texthero.org/docs/getting-started) document. I would start a bit gently by showing some examples and explain a bit deeper.
1. Before actually writing the content, we can just discuss the structure, the different headings, and sub-headings and the content of each sub-heading, without writing it?


Structure (just a draft)

#HeroSeries

One line intro to give an idea on the subject

## Overview

Explain the why
Show the 4 

## TextSeries

...

## TokenSeries

...

## RepresentationSeries

...

## VectorSeries

- mention better to avoid, used for `pca` as practical



Open question:

We can have either this page just after the "Getting Started" page or at the end.

Version 1:

1. Getting Started
1. Preprocessing
1. Representation
1. NLP
1. Visualization
1. HeroSeries

Version 2:

1. Getting Started
1. HeroSeries
1. Preprocessing
1. Representation
1. NLP
1. Visualization

We should consider which solution works best, particularly taking into consideration the content of the other pages.

The advantage of Version 1 is that the user understand immediately what HeroSeries are and stand for.

Advantage of Version 2 is that the user got to see the whole pipeline in the details and he should be more aware of the difference between the Series. On the other hand, it's also true that the "1. Getting Started" already provides a general introduction to the pipeline.

**Long story short, we need to have clear in mind the aim of this documentation page as well as how it will interface with the rest of the documentation. The idea again here is to provide a simple and unified view of Texthero.** 

I believe we can aim toward a results similar to the docusaurus documentation: [doc](https://docusaurus.io/docs/en/installation).

Extra:
- We need to explain clearly that these are just Pandas Series, nothing else

**Comment by henrifroese** [comment]
> Advantage of Version 2 is that the user got to see the whole pipeline in the details and he should be more aware of the difference between the Series. On the other hand, it's also true that the "1. Getting Started" already provides a general introduction to the pipeline.

We agree with this. Getting started provides an overview, then we have a deep-dive for the users really interested in the different modules through Preprocessing, Representation, NLP, Visualization; and if they still want to know more, they get to the HeroSeries stuff. I think many users will read the getting started, and then later on when googling they will land on `texthero.org` and then see the `TextSeries, RepresentationSeries, ...` in the docstrings. This will lead them to the tutorial too as we'll link it in the docstrings!

Also, we just improved the content / explained in more detail, and changed to your proposed structure.

File: website/docs/getting-started-herotypes.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+<h3 align="left">DataFrame</h3>
+
+In Natural Language Processing, we are often working with matrices that contain information about our dataset. For example, the output of the functions `tfidf`, `count`, and `term_frequency` is a [Document Term Matrix](https://en.wikipedia.org/wiki/Document-term_matrix), i.e. a matrix where each row is one document and each column is one term / word.
+
+We use a Pandas DataFrame for this for two reasons:
+1. It looks nice.
+2. It can be sparse.
+
+The second reason is worth explaining in more detail: In e.g. a big Document Term Matrix, we might have 10,000 different terms, so 10,000 columns in our DataFrame. Additionally, most documents will only contain a small subset of all the terms. Thus, in each row, there will be lots of zeros in our matrix. This is why we use a [sparse matrix](https://en.wikipedia.org/wiki/Sparse_matrix): A sparse matrix only stores the non-zero fields. And Pandas DataFrames support sparse data, so Texthero users fully profit from the sparseness!
```
Comment (mk2510) [inline website/docs/getting-started-herotypes.md:123]: Document-term and every where else as well

File: website/docs/getting-started-herotypes.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+This is a massive advantage when dealing with *big data*: In a _sparse DataFrame_, we only store the data that's relevant to save lots and lots of time and space!
```
Comment (mk2510) [inline website/docs/getting-started-herotypes.md:125]: maybe: big data sets, as with big data the user might think of something else

File: website/docs/getting-started-herotypes.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+3            "Text in the fourth document"
+4                                    ...
+
+```
+
+ Consequently, in Texthero's _preprocessing_ module, the functions usually take as input a Series where every cell is a string, and return as output a Series where every cell is a string. We will call this kind of Series _TextSeries_, so users know immediately what kind of Series the functions can work on. For example, you might see a function
+ ```python
+remove_punctuation(s: TextSeries) -> TextSeries
+ ```
+in the documentation. You then know that this function takes as input a _TextSeries_ and returns as output a _TextSeries_, so it can be used on a DataFrame or Series in the preprocessing phase of your work, where each document is one string.
```
Comment (mk2510) [inline website/docs/getting-started-herotypes.md:23]: I don't think it can be used on a dataframe as well, just on the textcolumn, which will then be interpreted as a series. So this sentence might be a bit confusing

File: website/docs/getting-started-herotypes.md (hunk 1 above)
Comment (henrifroese) [reply website/docs/getting-started-herotypes.md:123]: will do :paintbrush:

File: website/docs/getting-started-herotypes.md (hunk 2 above)
Comment (henrifroese) [reply website/docs/getting-started-herotypes.md:125]: done

File: website/docs/getting-started-herotypes.md (hunk 3 above)
Comment (henrifroese) [reply website/docs/getting-started-herotypes.md:23]: yes, you're right - will change :running_man:
